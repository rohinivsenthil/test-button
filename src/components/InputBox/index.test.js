import React from 'react';
import {InputBox} from './index';
import { render, fireEvent } from '@testing-library/react';

describe('the inputBox component', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<InputBox></InputBox>);

        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onChange function when text is typed into the input box', () => {
        const onChangeMock=jest.fn();
        const { getByTestId } = render(<InputBox testID="test-input" onChange={onChangeMock}></InputBox>);
        fireEvent.change(getByTestId("test-input"),{ target: { value:"Roh"}});
        expect(onChangeMock).toHaveBeenCalledWith("Roh");
    });

})
