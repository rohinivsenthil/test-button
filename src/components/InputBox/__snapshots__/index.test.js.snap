// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the inputBox component should render correctly 1`] = `
<DocumentFragment>
  <input
    type="text"
  />
</DocumentFragment>
`;
